gobblin-utility/src/main/java/gobblin/util/JobLauncherUtils.java

```diff
@@ -203,4 +218,12 @@ private static ParallelRunner getParallelRunner(FileSystem fs, Closer closer, in
     }
     return parallelRunners.get(uri);
   }
+
+  private static synchronized void updateOwnerAndFs(String owner, FileSystem proxiedFs) {
+    ownerAndFs.put(owner, proxiedFs);
+  }
+
+  private static synchronized FileSystem getFsWithProxyOwner(String owner) {
+    return ownerAndFs.get(owner);
+  }
```

Comment:
It may not be necessary to have these two methods since they are both just one line. Also, instead of making these methods synchronized, you can probably just use `Maps.newConcurrentMap`